### ISSUE TITLE
execl* argv list is missing

### ISSUE THREAD

**Issue by AsafEitani** [aquasecurity/tracee#724]
When running the following code, executing ls with `execl`, tracee-ebpf will not document the argv of the function.
I suspect that the reason being that any `execl*` function is getting the `argv` list as several string parameters, ending in null pointer, rather than a array of strings.

from `exec` man page:

```
       int execl(const char *pathname, const char *arg, ...
                       /* (char  *) NULL */);
       int execlp(const char *file, const char *arg, ...
                       /* (char  *) NULL */);
       int execle(const char *pathname, const char *arg, ...
                       /*, (char *) NULL, char *const envp[] */);

l - execl(), execlp(), execle()
       The const char *arg and subsequent ellipses can be thought of as arg0, arg1, ..., argn.  Together they describe a list of one or more pointers to null-terminated strings that represent the argument  list  available  to  the  executed  program.  The first argument, by convention, should point to the filename associated with the file being executed.  The list of arguments must be terminated by a null pointer, and, since these are variadic functions, this pointer must be cast (char *) NULL.

       By contrast with the 'l' functions, the 'v' functions (below) specify the command-line arguments of the executed program as a vector.
```

the C code: 
```
#include <stdio.h>
#include <unistd.h>

int main(void) {	
  execl("/bin/ls", "ls", "-l", NULL);
  return 1;
}
```

Tracee-ebpf output as json:

```
{
    "timestamp": 156707.711393,
    "processId": 213411,
    "threadId": 213411,
    "parentProcessId": 2592,
    "hostProcessId": 213411,
    "hostThreadId": 213411,
    "hostParentProcessId": 2592,
    "userId": 1000,
    "mountNamespace": 4026531840,
    "pidNamespace": 4026531836,
    "processName": "execl",
    "hostName": "ubuntu",
    "eventId": "59",
    "eventName": "execve",
    "argsNum": 1,
    "returnValue": 0,
    "stackAddresses": null,
    "args": [
        {
            "name": "argv",
            "type": "const char*const*",
            "value": null
        }
    ]
}
{
    "timestamp": 156707.712087,
    "processId": 213411,
    "threadId": 213411,
    "parentProcessId": 2592,
    "hostProcessId": 213411,
    "hostThreadId": 213411,
    "hostParentProcessId": 2592,
    "userId": 1000,
    "mountNamespace": 4026531840,
    "pidNamespace": 4026531836,
    "processName": "execl",
    "hostName": "ubuntu",
    "eventId": "1004",
    "eventName": "security_bprm_check",
    "argsNum": 3,
    "returnValue": 0,
    "stackAddresses": null,
    "args": [
        {
            "name": "pathname",
            "type": "const char*",
            "value": "/usr/bin/ls"
        },
        {
            "name": "dev",
            "type": "dev_t",
            "value": 8388611
        },
        {
            "name": "inode",
            "type": "unsigned long",
            "value": 3146398
        }
    ]
}
```

@itaysk :)

**Comment by yanivagman** [comment]
Actually, this "bug" has nothing to do with execl, and will also happen with the following program that is using execve:
```
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int
main(int argc, char *argv[])
{
    char *newargv[] = { NULL, "-l", NULL };
    char *newenviron[] = { NULL };

    newargv[0] = "ls";

    execve("/bin/ls", newargv, newenviron);
    return 1;
}
```

As you can see in the output of the execve event you added, not only argv is null, but also the pathname (which should be "/bin/ls") is missing!
When I first saw this bug, I noticed that it also happens with the execsnoop tool from bcc, so I opened a bug about it:
https://github.com/iovisor/bcc/issues/2627
In short, the command arguments strings are placed in the readonly section, and when using bpf_probe_read() to read the strings, it gets -EFAULT, which means that the page that contains the string was not fetched yet...

You can workaround this problem (only for the pathname and not the arguments) by using the security_bprm_check event. In the future, we might switch from kprobe/kretprobe to fentry/fexit, which might solve this problem.

**Comment by yanivagman** [comment]
As described above, this is not a bug of tracee.
We can discuss future options to workaround this in the discussions.
